engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -825,10 +826,14 @@ case class WorkflowActor(workflow: WorkflowDescriptor)
       case _ => false
     }
 
+    lazy val hasAnythingFailed = executionStore.values exists { x => x == Failed || x == Aborted }
+
+    lazy val workflowFailFastTrigger = workflowFailureMode == NoNewCalls && hasAnythingFailed
```

Comment:
it feels like most of this logic should live in the `WorkflowFailureMode` subtypes